tests/transforms/test_broadcast_expand.py

```diff
@@ -21,65 +21,50 @@
 from pennylane import numpy as pnp
 
 
-class RX_broadcasted(qml.RX):
-    """A version of qml.RX that detects batching."""
-
-    ndim_params = (0,)
-    compute_decomposition = staticmethod(lambda theta, wires=None: qml.RX(theta, wires=wires))
-
-
-class RZ_broadcasted(qml.RZ):
-    """A version of qml.RZ that detects batching."""
-
-    ndim_params = (0,)
-    compute_decomposition = staticmethod(lambda theta, wires=None: qml.RZ(theta, wires=wires))
-
-
 dev = qml.device("default.qubit", wires=2)
 """Defines the device used for all tests"""
 
 
-def make_tape(x, y, z, obs):
-    """Construct a tape with three parametrized, two unparametrized
-    operations and expvals of provided observables."""
-    with qml.queuing.AnnotatedQueue() as q:
-        qml.StatePrep(np.array([1, 0, 0, 0]), wires=[0, 1])
-        RX_broadcasted(x, wires=0)
-        qml.PauliY(0)
-        RX_broadcasted(y, wires=1)
-        RZ_broadcasted(z, wires=1)
-        qml.Hadamard(1)
-        for ob in obs:
-            qml.expval(ob)
-
-    tape = qml.tape.QuantumScript.from_queue(q)
-    return tape
-
-
-parameters_and_size = [
-    [(0.2, np.array([0.1, 0.8, 2.1]), -1.5), 3],
-    [(0.2, np.array([0.1]), np.array([-0.3])), 1],
-    [
-        (
-            0.2,
-            pnp.array([0.1, 0.3], requires_grad=True),
-            pnp.array([-0.3, 2.1], requires_grad=False),
-        ),
-        2,
-    ],
+def make_ops(x, y, z):
+    """Queue three parametrized and two unparametrized operations and return them."""
+    ops = [
+        qml.RX(x, wires=0),
+        qml.PauliY(0),
+        qml.RX(y, wires=1),
+        qml.RZ(z, wires=1),
+        qml.Hadamard(1),
+    ]
+    return ops
+
+
+parameters = [
+    (0.2, np.array([0.1, 0.8, 2.1]), -1.5),
+    (0.2, np.array([0.1]), np.array([-0.3])),
+    (
+        0.2,
+        pnp.array([0.1, 0.3], requires_grad=True),
+        pnp.array([-0.3, 2.1], requires_grad=False),
+    ),
 ]
+sizes = [3, 1, 2]
 
 coeffs0 = [0.3, -5.1]
 H0 = qml.Hamiltonian(qml.math.array(coeffs0), [qml.PauliZ(0), qml.PauliY(1)])
 
+
 # Here we exploit the product structure of our circuit
-exp_fn_Z0 = lambda x, y, z: -qml.math.cos(x) * qml.math.ones_like(y) * qml.math.ones_like(z)
-exp_fn_Y1 = lambda x, y, z: qml.math.sin(y) * qml.math.cos(z) * qml.math.ones_like(x)
+def exp_fn_Z0(x, y, z):
+    out = -qml.math.cos(x) * qml.math.ones_like(y) * qml.math.ones_like(z)
+    return out[0] if len(out) == 1 else out
+
+
+def exp_fn_Y1(x, y, z):
+    out = qml.math.sin(y) * qml.math.cos(z) * qml.math.ones_like(x)
+    return out[0] if len(out) == 1 else out
```

Comment:
For a batch dimension of `1`, the result is extracted by the PL execution pipeline. We need to mimic this here in the expected functions.